### PR TITLE
Create navbar dropdown

### PR DESCRIPTION
**Pull request by mxmeinhold**
Adds a dropdown to the navbar to contain logout and display settings.
Adds uid and profile image to the top right corner to match other CSH services.

[Preview](https://members.csh.rit.edu/~mom/Members/MaxMembers/)

### CODE REVIEW

File: templates/navbar.html [hunk 1]
```diff
@@ -1,18 +1,24 @@
 <!-- Navbar -->
-<nav class="navbar fixed-top navbar-expand-lg navbar-dark bg-primary">
+<nav class="navbar fixed-top navbar-expand-lg navbar-dark bg-primary" style="padding: 0 0">
```
Comment (mbillow) [inline templates/navbar.html]: You can’t remove the padding via a style tags here, it removes the padding on mobile.

File: templates/navbar.html [hunk 2]
```diff
@@ -1,18 +1,24 @@
...
+      <ul class="nav navbar-nav ml-auto my-2">
         <li class="nav-item">
-          <a class="nav-link" style="color: #fff" href="https://csh.rit.edu">Public Site</a>
+          <a class="nav-link btn-secondary bg-primary text-white" style="color: #fff" href="https://csh.rit.edu">Public Site</a>
         </li>
-        <li class="nav-item">
-          <a class="nav-link" style="color: #fff" href="https://members.csh.rit.edu/sso/logout">Logout</a>
+        <!-- Dropdown -->
+        <li class="nav-item navbar-user dropdown" >
+          <a class="nav-link dropdown-toggle bg-primary text-white mt-2" data-toggle="dropdown" href="#" id="nav-drop" aria-expanded="false">
```
Comment (mbillow) [inline templates/navbar.html]: Maybe a link to profiles?

**Comment by mbillow** [comment]
http://drop.billow.me/JQ1ppr

**Comment by RamZallan** [comment]
Looks like there's some weird styling on the element. If I click into the dropdown, then click ESC, it closes the dropdown but keeps my name highlighted:
image
whereas that's not a problem for sites w/ similar dropdowns like Conditional. You also may want to display `uid` vs `dn` (i.e. `ram` vs `Ram Zallan`), to be consistent w/ the one on Conditional and elsewhere, just a personal preference, though.

**Comment by stevenmirabito** [comment]
Yeah, as noted above, ensure you have the proper markup structure and classes applied to the navbar and its items as there are some weird style problems. Also, I'd suggest adding a separator to the dropdown as indicated below:

**Comment by mxmeinhold** [comment]
@RamZallan It seems as though the dropdown remaining focused is universal across all of our bootstrap 4 pages.

**Comment by mbillow** [comment]
Profiles handles everything correctly, for example.

**Comment by mxmeinhold** [comment]
The dropdown highlighting appears to be an artefact of Bootstrap 4.
Profiles and bootstrap's own examples both experience the same issue:
screen shot 2018-07-02 at 09 35 59 gmt-4
screen shot 2018-07-02 at 09 47 50 gmt-4

Both of these are dropdown triggers that are displayed as though active, even though the dropdown is not visible. Our pages using Bootstrap 3, such as STARRS and Conditional, do not display the toggles as active, even while they are capturing keyboard input (can be toggled with enter).

Having not found any bootstrap 4 solutions to this at present, for the present I am proceeding with the text highlight style of navbar buttons, resulting in near invisible button focus.

**Comment by natedooley** [comment]
Everything looks fine to me. Literally no one will ever run into this minuscule "issue" and even if they do no one cares. It's clearly a Bootstrap 4 issue anyway. Thumbs up from me.

**Pull request merged**